contracts/strategies/BaseStrategy.sol

```diff
@@ -23,21 +26,29 @@ abstract contract BaseStrategy is IStrategy {
     /// === Storage Variables ====
     /// ==========================
 
-    bytes32 internal STRATEGY_IDENTIFIER;
+    Allo public immutable allo;
 
-    bytes32 internal identityId;
-    Allo internal allo;
+    uint256 public poolId;
+    bytes32 public identityId;
 
-    uint256 internal poolId;
-    bool public initialized;
+    address public treasury;
 
     /// ====================================
-    /// =========== Modifier ===============
+    /// ========== Constructor =============
+    /// ====================================
+
+    /// @param _allo Address of the Allo contract
+    constructor(address _allo) {
+        allo = Allo(_allo);
```

Comment:
setting the allo contract on strategy deployment will make the contract unusable when we have to fully redeploy the Allo contract for some reason. -> we have to redeploy all strategies

On the other hand in this way the implementation is really only useable using Allo. 

Maybe we should discuss the pros and cons of both ways? @thelostone-mc @zobront @codenamejason @nfrgosselin

Reply:
What's the downside of doing it the earlier way ? Having it being set on the initialize function where it would be set to msg.sender ?

Reply:
@thelostone
Anyone can clone and use the implementation
That's the only downside I see